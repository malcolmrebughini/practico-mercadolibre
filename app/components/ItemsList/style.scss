@import 'app/styles/vars';

.itemsList {
  background-color: #fff;
  list-style: none;
  margin: 0 auto;

  @media (min-width: 768px) {
    max-width: $body-width;
    border-radius: 4px;
  }
}