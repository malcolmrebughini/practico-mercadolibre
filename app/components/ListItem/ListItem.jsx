import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import freeShipping from './assets/ic_shipping.png';
import styles from './styles.scss';


function ListItem(props) {
  const linkTo = {
    pathname: `/items/${props.id}`,
  };

  return (
    <li className={styles.item}>
      <Link title={props.title} to={linkTo}>
        <div className={styles.thumbnail}>
          <img src={props.thumbnail} alt={props.title} />
        </div>
        <div className={styles.description}>
          <span className={styles.price}>
            $ {props.price}
            {
              props.freeShipping &&
                <img alt="envio gratis" className={styles.freeShipping} src={freeShipping} />
            }
          </span>
          <span className={styles.title}>{props.title}</span>
        </div>
        <div className={styles.location}>
          {props.location}
        </div>
      </Link>
    </li>
  );
}


ListItem.propTypes = {
  id: PropTypes.string,
  thumbnail: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.number,
  location: PropTypes.string,
  freeShipping: PropTypes.bool,
};


export default ListItem;
