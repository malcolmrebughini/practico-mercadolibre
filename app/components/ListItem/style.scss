.item {
  padding: 16px;
  border-top: 1px solid rgba(51, 51, 51, 0.1);

  &:first-of-type {
    border-top: none;
  }

  a {
    text-decoration: none;
    display: flex;
    flex-direction: row;
    color: #333;

    &:visited {
      color: #333;
    }
  }

  .thumbnail {
    img {
      width: 180px;
      height: 180px;
      border-radius: 4px
    }
  }

  .description {
    padding-left: 16px;
    flex-grow: 2;

    .title {
      display: block;
      font-weight: 300;
      font-size: 14px;
      margin-top: 32px;

      @media (min-width: 768px) {
        font-size: 18;
        width: 300px;
      }
    }

    .price {
      display: block;
      font-size: 24px;
    }

    .freeShipping {
      margin-left: 10px;
    }
  }

  .location {
    display: none;

    @media (min-width: 768px) {
      display: block;
      flex-grow: 1;
      font-size: 12px;
      flex-basis: 100px;
    }
  }
}