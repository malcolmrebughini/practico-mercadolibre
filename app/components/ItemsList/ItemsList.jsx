import React from 'react';
import PropTypes from 'prop-types';
import ListItem from '../ListItem';
import styles from './styles.scss';


function ItemsList(props) {
  return (
    <ol className={styles.itemsList}>
      {props.items.map(item =>
        (<ListItem
          key={item.id}
          id={item.id}
          title={item.title}
          currencyId={item.currency_id}
          price={item.price}
          thumbnail={item.thumbnail}
          location={item.address.state_name}
          freeShipping={item.shipping.free_shipping}
        />),
      )
      }
    </ol>
  );
}


ItemsList.propTypes = {
  items: PropTypes.array,
};


ItemsList.defaultProps = {
  items: [],
};


export default ItemsList;
