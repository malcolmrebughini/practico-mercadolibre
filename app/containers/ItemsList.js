import { connect } from 'react-redux';
import ItemsList from '../components/ItemsList';


function mapStateToProps(state) {
  return {
    items: state.search.items,
  };
}

export default connect(mapStateToProps)(ItemsList);
