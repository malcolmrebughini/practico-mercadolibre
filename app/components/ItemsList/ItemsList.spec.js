import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ItemsList from './ItemsList';
import ListItem from '../ListItem';


const testItems = [
  { id: '1', address: {}, shipping: {} },
  { id: '2', address: {}, shipping: {} },
];


describe('ItemsList', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ItemsList items={testItems} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render a list of items', () => {
    const wrapper = shallow(<ItemsList items={testItems} />)
    const renderedItems = wrapper.find(ListItem);
    expect(renderedItems.length).toEqual(2);
  });
});
