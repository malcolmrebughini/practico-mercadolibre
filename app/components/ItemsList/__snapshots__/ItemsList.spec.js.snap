// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemsList should render correctly 1`] = `
<ol
  className="itemsList"
>
  <ListItem
    id="1"
  />
  <ListItem
    id="2"
  />
</ol>
`;
